### PR TITLE
New lmi path

### PR DESCRIPTION
**Pull request by thesiv**
Recreated https://github.com/let-me-illustrate/lmi/pull/156.

### CODE REVIEW

File: authenticity.cpp [hunk 1]
```diff
@@ -98,7 +96,10 @@ std::string Authenticity::Assay
     // Read saved passkey from file.
     std::string passkey;
     {
-    fs::path passkey_path(data_path / "passkey");
+    fs::path passkey_path("passkey");
+    // Add only non-empty data path to avoid adding "./".
+    if(!data_path.empty())
+        passkey_path = data_path / passkey_path;
```
Comment (vadz) [inline authenticity.cpp:102]: I wonder if it's worth customizing our `operator/()` behaviour to handle first argument being empty specially.

File: path.hpp [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+    :public std::ifstream
+{
+  public:
+    ifstream() = default;
+
+    explicit ifstream
+        (path const& p
+        ,std::ios_base::openmode mode = std::ios_base::in
+        )
+        :std::ifstream(static_cast<std::filesystem::path const&>(p), mode)
```
Comment (vadz) [inline path.hpp:440]: I really don't think we should/can do this. Let's provide an explicit accessor instead (or maybe use private inheritance and make this ctor friend of our `path`?).

**Comment by vadz** [comment]
FYI: I've rebased to the latest master (making some more changes to the code modified in the meanwhile), applied an ugly fix for C++20 build and force pushed to this branch.

**Comment by thesiv** [comment]
Rebased to `master`.

File: wx_test_output.hpp [hunk 3]
```diff
@@ -56,11 +56,6 @@ class output_file_existence_checker
         return fs::exists(path_);
         }
 
-    std::string const& path() const
```
Comment (vadz) [inline wx_test_output.hpp:60]: @thesiv Do you remember why did this commit have to be done?

It looks to me that just changing the return type here to `std::string` (i.e. returning by value) is enough to make the code compile and work and it seems preferable as it's a more minimal change.

**Comment by vadz** [comment]
Replaced with #364.

**Pull request closed without merging**